backend/typescript/services/implementations/animalTypeService.ts

```diff
@@ -0,0 +1,118 @@
+import PgAnimalType from "../../models/animalType.model";
+import {
+  IAnimalTypeService,
+  AnimalTypeRequestDTO,
+  AnimalTypeResponseDTO,
+} from "../interfaces/animalTypeService";
+import { getErrorMessage, NotFoundError } from "../../utilities/errorUtils";
+import logger from "../../utilities/logger";
+
+const Logger = logger(__filename);
+
+class AnimalTypeService implements IAnimalTypeService {
+  /* eslint-disable class-methods-use-this */
+  async getAnimalType(id: string): Promise<AnimalTypeResponseDTO> {
+    let animalType: PgAnimalType | null;
+    try {
+      animalType = await PgAnimalType.findByPk(id, { raw: true });
+      if (!animalType) {
+        throw new NotFoundError(`Animal type id ${id} not found`); // return json instead (for standardization of responses)?
```

Comment:
we're throwing in error, passing in a message so i think its okay to not have it return json